src/test/java/fr/openent/lystore/service/DefaultPurseServiceTest.java

```diff
@@ -0,0 +1,93 @@
+                "                                                      / 100 " +
+                "                                                    ) * " +
+                "                                                    oco.amount " +
+                "                                    ) IS " +
+                "                                    NULL THEN 0 " +
+                "                               ELSE Sum(Round(oco.price + ( " +
+                "                                              oco.price * oco.tax_amount " +
+                "                                              / 100 ) " +
+                "                                                          * " +
+                "                                        oco.amount, 2)) " +
+                "                             END " +
+                "                      FROM     " + Lystore.lystoreSchema + ".order_client_options oco " +
+                "                      WHERE  oco.id_order_client_equipment = oce.id) " +
+                "                     + Round((oce.price + oce.price * oce.tax_amount /100), 2) ) " +
+                "                   * " +
+                "                   oce.amount) " +
+                "               AS total_order " +
+                "        FROM     " + Lystore.lystoreSchema + ".order_client_equipment oce " +
+                "               INNER JOIN   " + Lystore.lystoreSchema + ".purse " +
+                "                       ON purse.id_campaign = oce.id_campaign " +
+                "                          AND oce.id_structure = purse.id_structure " +
+                "        GROUP  BY oce.id_structure, " +
+                "                  oce.id_campaign " +
+                "        ORDER  BY id_structure) " +
+                "SELECT purse.*, " +
+                "       orders.total_order " +
+                "FROM      " + Lystore.lystoreSchema + ".purse " +
+                "left join orders " +
+                "               ON orders.id_structure = purse.id_structure AND  orders.id_campaign = purse.id_campaign " +
+                "WHERE  " +
+                "        purse.id_campaign = ? ;  " ;
+
+        JsonArray expectedParams = new JsonArray();
+
+        vertx.eventBus().consumer("fr.openent.lystore", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        try {
+            Whitebox.invokeMethod(this.purseService, "getPursesByCampaignId",(Handler) e -> {
+            });
+        } catch (Exception e) {
+            ctx.assertNotNull(e);
+        }
```

Comment:
Vu que ta fonction appelle une promise, il vaut mieux que tu appelles directement la méthode Future pour déclencher ton test 

il te manque un param également sur ta méthode de Whitebox (ta méthode prend un Integer en param)

ton expectedParams vaut rien comme c'est juste une liste vide

Admettons que tu veuilles tester avec id_campaign en 5 :  

```suggestion
                  JsonArray expectedParams = new JsonArray().add(5);
        
        vertx.eventBus().consumer("fr.openent.lystore", message -> {
            JsonObject body = (JsonObject) message.body();
            ctx.assertEquals("prepared", body.getString("action"));
            ctx.assertEquals(expectedQuery, body.getString("statement"));
            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
            async.complete();
        });

        this.purseService.getPursesByCampaignId(5);
```